xcm/xcm-builder/src/barriers.rs

```diff
@@ -65,7 +65,13 @@ impl<T: Contains<MultiLocation>> ShouldExecute for AllowTopLevelPaidExecutionFro
 		);
 		ensure!(T::contains(origin), ());
 		let mut iter = message.0.iter_mut();
-		let i = iter.next().ok_or(())?;
+		let mut i = iter.next().ok_or(())?;
+		match i {
+			SetAppendix(Xcm(ref instrs)) if matches!(&instrs[..], &[ReportError { .. }]) => {
```

Comment:
This here is basically ignoring the `SetAppendix` instruction if it appears before an asset transfer instruction. The one thing that I am concerned about is whether the `SetAppendix` instruction is properly paid for, or whether it's even necessary to pay for it. I will likely need to defer to @gavofyork for this change.

Reply:
Also, it might also be good to include `SetErrorHandler` as well.

Reply:
Er... no. This is super dangerous. You would just do a `SetAppendix(<huge amount of code>)` followed by a `BuyExecution` which fails. You'd then get all the huge amount of code executed for free.

Reply:
`matches!(&instrs[..], &[ReportError { .. }])` actually only allows one instruction which is `ReportError`